body/stretch_body/stepper.py

```diff
@@ -478,12 +480,13 @@ def read_encoder_calibration_from_YAML(self):
         enc_data=read_fleet_yaml(fn)
         return enc_data
 
-    def write_encoder_calibration_to_YAML(self,data):
+    def write_encoder_calibration_to_YAML(self,data,filename=None, fleet_dir=None):
         device_name = self.usb[5:]
-        sn = self.robot_params[device_name]['serial_no']
-        fn = 'calibration_steppers/'+device_name + '_' + sn + '.yaml'
-        print('Writing encoder calibration: %s'%fn)
-        write_fleet_yaml(fn,data)
+        if filename is None:
+            sn = self.robot_params[device_name]['serial_no']
+            fn = 'calibration_steppers/'+device_name + '_' + sn + '.yaml'
+        print('Writing encoder calibration: %s'%filename)
+        write_fleet_yaml(filename,data,fleet_dir=fleet_dir)
```

Comment:
Won't this cause a bug if `filename` is None? `filename` is never set equal to `fn`.

Reply:
Good catch, thanks. The `fn` variable is renamed to `filename` which should work.